### PR TITLE
Fix Graph Break on builtin comparison on NNModule

### LINKED ISSUE #102338: [Dynamo] Unsupported: comparison NNModuleVariable() <built-in function eq> NNModuleVariable()
### 🐛 Describe the bug

Repro:
```
import torch
import torch._dynamo
import logging
import torch.nn.functional as F

torch._logging.set_logs(dynamo=logging.DEBUG)

class MyModule(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.layer0 = torch.nn.Linear(10, 10)
        self.layer1 = torch.nn.Linear(10, 10)
        self.layer2 = torch.nn.Linear(10, 10)

    @property
    def encoder_layers(self):
        return [self.layer0, self.layer1, self.layer2]

    def forward(self, x):
        for layer in self.encoder_layers:
            output = layer(x)
            if layer == self.layer0:
                output = F.relu6(output)
            else:
                output = F.relu(output)
        return output


x = torch.randn(10, 10)

m = MyModule()
print(m(x))

opt_m = torch.compile(backend="eager")(m)
print(opt_m(x))
```

Error:
```
Traceback (most recent call last):
  File "/scratch/ybliang/work/repos/pytorch/torch/_dynamo/symbolic_convert.py", line 663, in step
    getattr(self, inst.opname)(inst)
  File "/scratch/ybliang/work/repos/pytorch/torch/_dynamo/symbolic_convert.py", line 1082, in COMPARE_OP
    BuiltinVariable(supported_any[op], **options).call_function(
  File "/scratch/ybliang/work/repos/pytorch/torch/_dynamo/variables/builtin.py", line 586, in call_function
    result = handler(tx, *args, **kwargs)
  File "/scratch/ybliang/work/repos/pytorch/torch/_dynamo/variables/builtin.py", line 1276, in _comparison
    _unimplemented()
  File "/scratch/ybliang/work/repos/pytorch/torch/_dynamo/variables/builtin.py", line 1230, in _unimplemented
    unimplemented(f"comparison {typestr(left)} {op} {typestr(right)}")
  File "/scratch/ybliang/work/repos/pytorch/torch/_dynamo/exc.py", line 134, in unimplemented
    raise Unsupported(msg)
torch._dynamo.exc.Unsupported: comparison NNModuleVariable() <built-in function eq> NNModuleVariable()
```

### Versions

N/A

cc @ezyang @msaroufim @wconstab @ngimel @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

### PR DESCRIPTION
**Pull request by msaroufim**
Fixes https://github.com/pytorch/pytorch/issues/102338

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy

### CODE REVIEW

File: torch/_dynamo/symbolic_convert.py [hunk 1]
```diff
@@ -1060,6 +1060,17 @@ def COMPARE_OP(self, inst):
                     supported_const_comparison_ops[op](object(), right.value), **options
                 )
             )
+        elif(
+            isinstance(left, NNModuleVariable)
+            and isinstance(right, NNModuleVariable)
+            and op in supported_const_comparison_ops
```
Comment (msaroufim) [inline torch/_dynamo/symbolic_convert.py:1066]: should probably make this just `op in ("==)` for now

**Review by anijain2305 — APPROVED** [review]

**Pull request closed without merging**